### PR TITLE
Preserve request image files in logs directory

### PR DESCRIPTION
**Pull request by leonsim**
(Thanks for sending a pull request! Please make sure to read the contribution guidelines, then fill out the blanks below.)

What changes were proposed in this pull request?
------------------------------------------------

Does this close any currently open issues?
------------------------------------------
#283 

How was this patch tested?
--------------------------

### CODE REVIEW

File: bentoml/server/bento_api_server.py [hunk 1]
```diff
@@ -195,21 +194,54 @@ def bento_service_api_wrapper(api, service_name, service_version):
...
-                        "response": response.response,
-                    }
-                )
+
+            request_log = {
+                "uuid": request_id,
+                "service_name": service_name,
+                "service_version": service_version,
+                "api": api.name,
+                "request": _request_to_json(request),
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: in the case of image request, can we add the path to saved files here?

**Comment by parano** [comment]
looks like we also want to turn off this feature in tests environment, I did something similar in this PR here https://github.com/bentoml/BentoML/pull/287

File: bentoml/server/bento_api_server.py [hunk 2]
```diff
@@ -195,21 +194,54 @@ def bento_service_api_wrapper(api, service_name, service_version):
     summary_name = str(service_name) + "_" + str(api.name)
     request_metric_time = Summary(summary_name, summary_name + " request latency")
 
+    def log_image(request, request_id):
+        img_prefix = 'image/'
+        log_folder = config['logging'].get('base_log_folder')
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: another change after https://github.com/bentoml/BentoML/pull/287 is to use config('logging').get('base_log_folder')

Comment (leonsim) [reply bentoml/server/bento_api_server.py]: Ok, I'll rebase and update.

File: bentoml/server/bento_api_server.py (hunk 1 above)
Comment (leonsim) [reply bentoml/server/bento_api_server.py]: good idea. will do.

File: bentoml/server/bento_api_server.py [hunk 3]
```diff
@@ -194,21 +193,66 @@ def bento_service_api_wrapper(api, service_name, service_version):
...
+            request_log = {
+                "uuid": request_id,
+                "service_name": service_name,
+                "service_version": service_version,
+                "api": api.name,
+                "request": _request_to_json(request),
+                "response_code": response.status_code,
+            }
+
+            if image_paths is not None:
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: image_paths is [] for normal json request right?

File: bentoml/server/bento_api_server.py [hunk 4]
```diff
@@ -194,21 +193,66 @@ def bento_service_api_wrapper(api, service_name, service_version):
...
+            path = os.path.join(log_folder, filename)
+            all_paths.append(path)
+            with open(path, 'w') as f:
+                f.write(request.get_data())
+
+        for name in request.files:
+            file = request.files[name]
+            if file and file.filename:
+                orig_filename = secure_filename(file.filename)
+                filename = '{request_id}-{timestamp}-{orig_filename}'.format(
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: consider '{timestamp}-{request_id}-{orig_filename}'? so when doing manual debugging, image files will be sorted nicely by time?

File: bentoml/server/bento_api_server.py [hunk 5]
```diff
@@ -194,21 +193,66 @@ def bento_service_api_wrapper(api, service_name, service_version):
...
-                        "service_name": service_name,
-                        "service_version": service_version,
-                        "api": api.name,
-                        "request": _request_to_json(request),
-                        "response": response.response,
-                    }
-                )
+
+            request_log = {
+                "uuid": request_id,
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: could you help rename this field to just request_id?  `"request_id": request_id`

File: bentoml/server/bento_api_server.py [hunk 6]
```diff
@@ -59,12 +62,8 @@ def _request_to_json(req):
     """
     if req.content_type == "application/json":
         return req.get_json()
-    elif "image" in req.content_type:
-        return {"data": "dont handle"}
-    elif "video" in req.content_type:
-        return {"data": "dont handle"}
 
-    return {"data": req.get_data().decode("utf-8")}
+    return {"data": "ignore content type other than JSON."}
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: just `return {}`? 

for a image handler api server, we don't want to pollute the logs with this message `{"data": "ignore content type other than JSON."}` right?

Comment (leonsim) [reply bentoml/server/bento_api_server.py]: Yes, since we have output the file paths, this won't be needed.

**Pull request merged**